File: crates/crystallize-core/src/system/network.rs
use crate::system::fs;
use crate::utils::eval::files_eval;

pub fn set_hostname(hostname: &str) {
    println!("Setting hostname to {hostname}");
    fs::create_file("/mnt/etc/hostname");
    files_eval(
        fs::append_file("/mnt/etc/hostname", hostname),
        "set hostname",
    );
}

pub fn create_hosts() {
    fs::create_file("/mnt/etc/hosts");
    files_eval(
        fs::append_file("/mnt/etc/hosts", "127.0.0.1     localhost"),
        "create /etc/hosts",
    );
}

pub fn enable_ipv6() {
    files_eval(
        fs::append_file("/mnt/etc/hosts", "::1 localhost"),
        "add ipv6 localhost",
    );
}

File: internal/modules/network/network.go
package network

import "crystallize-cli/internal/utils"

// SetHostname sets the system hostname
func SetHostname(hostname string) {
	utils.LogInfo("Setting hostname to %s", hostname)
	_ = utils.CreateFile("/mnt/etc/hostname")
	utils.FilesEval(utils.AppendFile("/mnt/etc/hostname", hostname), "set hostname")
}

// CreateHosts creates the /etc/hosts file
func CreateHosts() {
	_ = utils.CreateFile("/mnt/etc/hosts")
	utils.FilesEval(
		utils.AppendFile("/mnt/etc/hosts", "127.0.0.1     localhost"),
		"create /etc/hosts",
	)
}

// EnableIPv6 enables IPv6 localhost entry
func EnableIPv6() {
	utils.FilesEval(
		utils.AppendFile("/mnt/etc/hosts", "::1 localhost"),
		"add ipv6 localhost",
	)
}
